File: apps/auth-service/app/src/main/java/ml/ipvz/fa/authservice/client/UserServiceClient.java
package ml.ipvz.fa.authservice.client;

import ml.ipvz.fa.authservice.model.dto.LoginDto;
import ml.ipvz.fa.authservice.model.dto.UserDto;
import reactor.core.publisher.Mono;

public interface UserServiceClient {
    Mono<UserDto> login(Mono<LoginDto> loginDto);
}

File: apps/user-service/client/src/main/kotlin/ml/ipvz/fa/userservice/client/UserServiceClient.kt
package ml.ipvz.fa.userservice.client

import ml.ipvz.fa.authservice.base.permission.Permission
import ml.ipvz.fa.userservice.model.LoginDto
import ml.ipvz.fa.userservice.model.UpdatePermissionsDto
import ml.ipvz.fa.userservice.model.UserDto
import org.springframework.http.ResponseEntity
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

interface UserServiceClient {
    fun login(loginDto: Mono<LoginDto>): Mono<UserDto>
    fun getUser(id: Long): Mono<UserDto>
    fun updatePermissions(updates: List<UpdatePermissionsDto>): Mono<ResponseEntity<Void>>
    fun getPermissions(userId: Long): Flux<Permission>
}